### PR TITLE
Port notebook 3 to PyTorch

### PR DESCRIPTION
**Pull request by ashwinvis**
TODO:
- [x] Check conventions on function names for preparing, training (we call it `training_loop`??) -> called it `fit` at the end. See https://docs.pytorch.org/tutorials/beginner/nn_tutorial.html#create-fit-and-get-data
- [x] Rename criterion -> loss_fn
- [x] Refactor valid part to be reused in eval
- [x] Better printing which yields RMSE too and possibly something more neat? See if **torchmetrics** helps, and also introduce **tqdm**.
    - use sklearn.metrics and array-api support
- [x] Fix names of layers. There is no "input_layer", and move batchnorm to the top
- [x] Compare with baseline model after every improvement. Can also do a scatter plot with the baseline.
- [x] Visualization with black diagonal line
- [ ] Test Tensorboard
- [ ] Mention some easy automated frameworks like PyTorch Lightning. And other options for "experiment tracking" that tensorboard such as wandb.
- [ ] Integrate exercises

### CODE REVIEW

**Review by otaub — APPROVED** [review]
I would pass the device as a parameter to the train_epoch function not use globals, but that's probably a matter of taste at this stage.

**Comment by ashwinvis** [comment]
Good point. I can fix that.
